examples/demo-rollup-avail/src/ledger_rpc.rs

```diff
@@ -0,0 +1,80 @@
+use jsonrpsee::RpcModule;
+use serde::de::DeserializeOwned;
+use serde::Serialize;
+use sov_db::ledger_db::LedgerDB;
+use sov_rollup_interface::rpc::{
+    BatchIdentifier, EventIdentifier, LedgerRpcProvider, SlotIdentifier, TxIdentifier,
+};
+
+use self::query_args::{extract_query_args, QueryArgs};
+
+/// Registers the following RPC methods
+/// - `ledger_head`
+///    Example Query: `curl -X POST -H "Content-Type: application/json" -d '{"jsonrpc":"2.0","method":"ledger_head","params":[],"id":1}' http://127.0.0.1:12345`
+/// - ledger_getSlots
+///    Example Query: `curl -X POST -H "Content-Type: application/json" -d '{"jsonrpc":"2.0","method":"ledger_getSlots","params":[[1, 2], "Compact"],"id":1}' http://127.0.0.1:12345`
+/// - ledger_getBatches
+///    Example Query: `curl -X POST -H "Content-Type: application/json" -d '{"jsonrpc":"2.0","method":"ledger_getBatches","params":[[1, 2], "Standard"],"id":1}' http://127.0.0.1:12345`
+/// - ledger_getTransactions
+///    Example Query: `curl -X POST -H "Content-Type: application/json" -d '{"jsonrpc":"2.0","method":"ledger_getBatches","params":[[1, 2], "Full"],"id":1}' http://127.0.0.1:12345`
+/// - ledger_getEvents
+///    Example Query: `curl -X POST -H "Content-Type: application/json" -d '{"jsonrpc":"2.0","method":"ledger_getBatches","params":[1, 2],"id":1}' http://127.0.0.1:12345`
+fn register_ledger_rpc_methods<B: Serialize + DeserializeOwned, T: Serialize + DeserializeOwned>(
+    rpc: &mut RpcModule<LedgerDB>,
+) -> Result<(), jsonrpsee::core::Error> {
+    rpc.register_method("ledger_getHead", move |_, db| {
+        db.get_head::<B, T>().map_err(|e| e.into())
```

Comment:
We can replace this and following `map_err` with `?` since `?` is doing implicite `into`